doc/advanced/content/executor_design.rst

```diff
@@ -346,74 +342,73 @@ Best Fit Executor
 =================================
 
 In most scenarios users of PCL do not care much about performance and simply want their code to run
-fast without any additional steps performed from their side. For this reason their was a need for
-mechanism called best fit, which automatically selects best possible executor and its customizes its
+fast without any additional steps performed from their side. For this reason, there was a need for a
+a mechanism called the best fit, which automatically selects the best possible executor and customizes its
 property in order to give the user good performance out of the box based on their system configurations
 and the function they are calling.
 
-The mechanism works be choosing an executor based on the availability of an executor (depends on
-hardware/software of a system) and the priority of executors specified in function by PCL
-maintainers & contributors which will give good performance. The executor properties can also be
-customized on for a better fit for certain scenarios. Besides this runtime checks are also be specified
-and on the basis of these runtime checks the executors are further filtered to selected most appropriate
-one. The two supported mechanisms in PCL currently are `enable_exec_with_priority` and
+The mechanism works by choosing an executor based on its availability (depends on
+hardware/software of a system) and its priority specified in function by PCL
+maintainers & contributors, which will give a good performance. Executor properties can also be customized
+to better fit certain scenarios. Besides these runtime checks are also specified, and on the basis
+of these runtime checks, the executors are further filtered to select the most appropriate
+one. The two supported mechanisms in PCL currently there are `enable_exec_with_priority` and
```

Comment:
This last sentence still reads weird. Currently there are two supported mechanisms...

Reply:
Reworded